lib/bolt/task.rb

```diff
@@ -8,30 +8,33 @@ def initialize(target, task)
     end
   end
 
-  # Represents a Task.
-  # @file and @files are mutually exclusive.
-  # @name [String] name of the task
-  # @file [Hash, nil] containing `filename` and `file_content`
-  # @files [Array<Hash>] where each entry includes `name` and `path`
-  # @metadata [Hash] task metadata
-  Task = Struct.new(
-    :name,
-    :file,
-    :files,
-    :metadata
-  ) do
-    attr_reader :remote
-
-    def initialize(task, remote: false)
-      super(nil, nil, [], {})
-
+  class Task
+    METADATA_KEYS = %w[description extensions files implementations
+                       input_method parameters private puppet_task_version
+                       remote supports_noop].freeze
+
+    attr_reader :name, :files, :metadata, :remote
+
+    # name [String] name of the task
+    # files [Array<Hash>] where each entry includes `name` and `path`
+    # metadata [Hash] task metadata
+    def initialize(name, metadata = {}, files = [], remote = false)
+      @name = name
+      @metadata = metadata || {}
+      @files = files || []
```

Comment:
I don't think the conditional assignment is needed since the parameters have defaults.

Reply:
That's mostly for the `from_task_signature` path, where it will pass `nil` if the keys aren't present in the hash. I can add these defaults there instead.